src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -257,89 +257,162 @@ default MemorySegment allocateFrom(AddressLayout layout, MemorySegment value) {
         return segment;
     }
 
+    /**
+     * Allocates a memory segment with the given layout and initializes it with the bytes in the provided
+     * source memory segment.
+     * @implSpec the default implementation for this method calls {@code this.allocate(elementLayout, elementCount)}.
+     * @param elementLayout the value to be set on the newly allocated memory block.
+     * @param source the source segment.
+     * @param sourceElementLayout the element layout of the source segment.
+     * @param sourceOffset the starting offset, in bytes, of the source segment.
+     * @param srcElementCount the number of elements in the source segment to be copied.
```

Comment:
Maybe just `elementCount` here. It doesn't necessarily have any connection to the source? (`elementCount` is also used in the IOOBE `@throws` clauses